computations.py

```diff
@@ -0,0 +1,140 @@
+
+    @staticmethod
+    def find_average_highest_temperature_in_month(year, month, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        maximum_temperature = Computations.get_weather_attribute_index(weather_attributes, "Max TemperatureC")
+        average_highest_temperature_in_month = 0
+        try:
+            no_of_weather_records_in_month = len(weather_records[year][month])
+            for single_day_record in weather_records[year][month]:
+                if single_day_record[maximum_temperature]:
+                    average_highest_temperature_in_month += single_day_record[maximum_temperature]
+            return average_highest_temperature_in_month // no_of_weather_records_in_month
+        except KeyError:
+            sys.exit('Weather Record not exist for {} {}'.format(year, calendar.month_name[month]))
+
+    @staticmethod
+    def find_average_lowest_temperature_in_month(year, month, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        minimum_temperature = Computations.get_weather_attribute_index(weather_attributes, "Min TemperatureC")
+        average_lowest_temperature_in_month = 0
+        try:
+            no_of_weather_records_in_month = len(weather_records[year][month])
+            for single_day_record in weather_records[year][month]:
+                if single_day_record[minimum_temperature]:
+                    average_lowest_temperature_in_month += single_day_record[minimum_temperature]
+            return average_lowest_temperature_in_month // no_of_weather_records_in_month
+        except KeyError:
+            sys.exit('Weather Record not exist for {} {}'.format(year, calendar.month_name[month]))
+
+    @staticmethod
+    def find_average_mean_humidity_in_month(year, month, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        mean_humidity = Computations.get_weather_attribute_index(weather_attributes, " Mean Humidity")
+        average_mean_humidity_in_year = 0
+        try:
+            no_of_weather_records_in_month = len(weather_records[year][month])
+            for single_day_record in weather_records[year][month]:
+                if single_day_record[mean_humidity]:
+                    average_mean_humidity_in_year += single_day_record[mean_humidity]
+            return average_mean_humidity_in_year // no_of_weather_records_in_month
+        except KeyError:
+            sys.exit('Weather Record not exist for {} {}'.format(year, calendar.month_name[month]))
+
+    @staticmethod
+    def find_maximum_temperature_on_day(year, month, day, all_weather_records):
+        weather_attributes, weather_records = all_weather_records
+        maximum_temperature = Computations.get_weather_attribute_index(weather_attributes, "Max TemperatureC")
+        try:
+            return weather_records[year][month][day][maximum_temperature]
```

Comment:
It's not a good practice to return directly from `try` block. You should get that input and then use `finally` and return it gracefully.